### PR TITLE
fix: return the correct delta result in `sliceToDelta`

### LINKED ISSUE #601: Incorrect copied result with more than one delta operation
## Description
It's a bug of the [`sliceToDelta`](https://github.com/toeverything/blocksuite/blob/3543ce264be856958312a38b0165e158a326558a/packages/store/src/text-adapter.ts#LL233C8-L233C8) method. When more than one delta operation was copied, the `contentText` variable of the last operation was [updated](https://github.com/toeverything/blocksuite/blob/3543ce264be856958312a38b0165e158a326558a/packages/store/src/text-adapter.ts#L250) without being [pushed](https://github.com/toeverything/blocksuite/blob/3543ce264be856958312a38b0165e158a326558a/packages/store/src/text-adapter.ts#L261) into `result`.

## Record
eUQjoMvZF5

### PR DESCRIPTION
**Pull request by donaldxdonald**
fixed #601

### CODE REVIEW

File: packages/store/src/text-adapter.ts [hunk 1]
```diff
@@ -231,38 +231,43 @@ export class Text {
...
       return delta;
     }
-    const result = [];
+
     if (delta && delta instanceof Array) {
       let charNum = 0;
       for (let i = 0; i < delta.length; i++) {
-        const content = delta[i];
-        let contentText = content.insert || '';
+        const content: DeltaOperation = delta[i];
```
Comment (himself65) [inline packages/store/src/text-adapter.ts:247]:
```suggestion
        const content = delta[i];
```

**Review by doodlewind — COMMENTED** [review]
Nice catch! Looks like this case can be easily reproduced as a test case, would you like to give it a try? Otherwise, there may be regressions in our future refactoring.

File: packages/store/src/text-adapter.ts (hunk 1 above)
Comment (donaldxdonald) [reply packages/store/src/text-adapter.ts:247]: I think using `any` everywhere is not a good idea. 🤔

Comment (himself65) [reply packages/store/src/text-adapter.ts:247]: sorry if misleading, I mean make `toDelta` to

```ts
toDelta(): DeltaOperation[] {
```

**Pull request merged**